feat: add ability to change location of cache files (#34337)
electron/electron: docs/api/app.md

    * `$XDG_CONFIG_HOME` or `~/.config` on Linux
    * `~/Library/Application Support` on macOS
  * `userData` The directory for storing your app's configuration files, which

>     by default is the `appData` directory appended with your app's name. Some
>     environments may synchronizes this directory to cloud.

Reviewer: ```suggestion
    by default is the `appData` directory appended with your app's name. Some
    environments may synchronize this directory to the cloud. For more information,
    read [where to store app data](../tutorial/where-to-store-app-data.md).
```

  * `localUserData` Similar to `userData` for storing configuration files. On
    Linux and macOS it is an alias to `userData`, on Windows it points to
    `%LOCALAPPDATA%` directory appended with your app's name. On Windows this
    directory will not be synchronized to cloud.
  * `cache` Per-user cache directory, which by default points to:
    * `%LOCALAPPDATA%` on Windows
    * `$XDG_CACHE_HOME` or `~/.cache` on Linux
    * `~/Library/Caches` on macOS
  * `userCache` The directory for storing your app's cache files, which
    by default is the `cache` directory appended with your app's name. HTTP disk

>     cache and JavaScript code cache are written into this directory.

Author: The name follows the name convention of `appData`/`userData` paths, which came from the `DIR_APP_DATA`/`DIR_USER_DATA` paths in Chromium, and which very likely came from the `%APPDATA%` env on Windows.

You can only override paths of a `name` defined in `app.getPath`.

By default, web pages' HTTP caches will be written to the `userCache` directory,
and other files like cookies and localStorage will be stored under the
`sessionData` directory. If you want to change this location, you have to
override the paths before the `ready` event of the `app` module is emitted.

> For most apps it is recommended to change the location of `sessionData`
> directory, for more please read
> [where to store app data](../tutorial/where-to-store-app-data.md).

Reviewer: ```suggestion
Most apps should change the location of `sessionData`. For more information, please
read [where to store app data](../tutorial/where-to-store-app-data.md).
```
